tests/core/arch-arm64/stalker-arm64.c

```diff
@@ -0,0 +1,660 @@
+  g_mutex_unlock (&ctx.mutex);
+
+  g_assert_cmpuint (fixture->sink->events->len, >, 0);
+
+  /* 8: Unfollow and notify victim about it */
+  gum_stalker_unfollow (fixture->stalker, thread_id);
+  g_mutex_lock (&ctx.mutex);
+  ctx.state = STALKER_VICTIM_IS_UNFOLLOWED;
+  g_cond_signal (&ctx.cond);
+  g_mutex_unlock (&ctx.mutex);
+
+  /* 9: Wait for victim to tell us it's ready for us to reset the sink */
+  g_mutex_lock (&ctx.mutex);
+  while (ctx.state != STALKER_VICTIM_READY_FOR_SHUTDOWN)
+    g_cond_wait (&ctx.cond, &ctx.mutex);
+  g_mutex_unlock (&ctx.mutex);
+
+  gum_fake_event_sink_reset (fixture->sink);
+
+  /* 12: Tell victim to it' */
+  g_mutex_lock (&ctx.mutex);
+  ctx.state = STALKER_VICTIM_IS_SHUTDOWN;
+  g_cond_signal (&ctx.cond);
+  g_mutex_unlock (&ctx.mutex);
+
+  g_thread_join (thread);
+
+  g_assert_cmpuint (fixture->sink->events->len, ==, 0);
+
+  g_mutex_clear (&ctx.mutex);
+  g_cond_clear (&ctx.cond);
+}
+
+STALKER_TESTCASE (heap_api)
+{
+  gpointer p;
+
+  fixture->sink->mask = (GumEventType) (GUM_EXEC | GUM_CALL | GUM_RET);
+
+  gum_stalker_follow_me (fixture->stalker, GUM_EVENT_SINK (fixture->sink));
+  p = malloc (1);
+  free (p);
+  gum_stalker_unfollow_me (fixture->stalker);
+
+  g_assert_cmpuint (fixture->sink->events->len, >, 0);
+}
+
+typedef void (* ClobberFunc) (GumCpuContext * ctx);
+
+STALKER_TESTCASE (no_register_clobber)
+{
+  guint8 * code;
+  GumArm64Writer cw;
+  const gchar * my_func_lbl = "my_func";
+  const gchar * my_beach_lbl = "my_beach";
+  const gchar * my_ken_lbl = "my_ken";
+  gint i;
+  gint offset;
+  ClobberFunc func;
+  GumCpuContext ctx;
+
+  code = gum_alloc_n_pages (1, GUM_PAGE_RWX);
+  gum_arm64_writer_init (&cw, code);
+
+  gum_arm64_writer_put_push_all_x_registers (&cw);  /* 16 push of 16 */
+
+  gum_arm64_writer_put_push_all_x_registers (&cw);
+  gum_arm64_writer_put_call_address_with_arguments (&cw,
+      gum_stalker_follow_me, 2,
+      GUM_ARG_ADDRESS, fixture->stalker,
+      GUM_ARG_ADDRESS, fixture->sink);
+
+  gum_arm64_writer_put_pop_all_x_registers (&cw);
+
+  for (i = ARM64_REG_X0; i <= ARM64_REG_X28; i++)
+  {
+    gum_arm64_writer_put_ldr_reg_u64 (&cw, i, i);
+  }
+
+  gum_arm64_writer_put_push_all_x_registers (&cw);
+  gum_arm64_writer_put_call_address_with_arguments (&cw,
+      gum_stalker_unfollow_me, 1,
+      GUM_ARG_ADDRESS, fixture->stalker);
+  gum_arm64_writer_put_pop_all_x_registers (&cw);
+
+  offset = (4 * sizeof (gpointer)) + (32 * sizeof (gpointer));
+
+  for (i = ARM64_REG_X0; i <= ARM64_REG_X28; i++)
+  {
+    gum_arm64_writer_put_str_reg_reg_offset (&cw, i, ARM64_REG_SP,
+        offset + G_STRUCT_OFFSET (GumCpuContext, x[i - ARM64_REG_X0]));
+  }
+
+  gum_arm64_writer_put_pop_all_x_registers (&cw);
+
+  gum_arm64_writer_put_ret (&cw);
+
+  gum_arm64_writer_free (&cw);
+
+  fixture->sink->mask = GUM_CALL | GUM_RET | GUM_EXEC;
+  func = GUM_POINTER_TO_FUNCPTR (ClobberFunc, code);
+  func (&ctx);
+
+  for (int i = ARM64_REG_X0; i <= ARM64_REG_X28; i++) {
```

Comment:
^ This one guy just doesn't want to die ;)

To reiterate:
- `int` shouldn't be declared here (declaration after statement), and `i` already exists in this scope anyway.
- Opening brace is on the wrong line.